compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -176,12 +182,12 @@ impl<'a, 'tcx> Encodable<EncodeContext<'a, 'tcx>> for SyntaxContext {
 
 impl<'a, 'tcx> Encodable<EncodeContext<'a, 'tcx>> for ExpnId {
     fn encode(&self, s: &mut EncodeContext<'a, 'tcx>) -> opaque::EncodeResult {
-        rustc_span::hygiene::raw_encode_expn_id(
-            *self,
-            &s.hygiene_ctxt,
-            ExpnDataEncodeMode::Metadata,
-            s,
-        )
+        if self.krate == LOCAL_CRATE {
+            // We will only write details for local expansions.
```

Comment:
That's true, but why?
What happens if we have a foreign expansion here and don't write details for it?

Reply:
In that case, the decoder will fetch the expansion from the other crate's metadata.

Reply:
Aha, finally found that issue - https://github.com/rust-lang/rust/issues/73047 (and https://github.com/rust-lang/rust/issues/74731).

Sometimes it's preferable for the "other crate" to not exist in its `.rlib` form, proc macro crates and other "macro only" crates clearly "overstay their welcome" when they exist only for the spans to be decodable from them.
I also know that right now we simply drop some spans from proc macro (dependency?) crates so their rlibs are not required, but ideally they would be preserved.

So eventually we'll need some span metadata inlining scheme (which includes hygiene data).

Reply:
Could you add a comment saying that here we are relying on the other crate's metadata existing, and maybe linking to these issues?